pkg/pillar/dpcmanager/verify.go

```diff
@@ -320,6 +322,30 @@ func (m *DpcManager) verifyDPC(ctx context.Context) (status types.DPCState) {
 		return status
 	}
 
+	// Check for the readiness of uplink ports as signaled by the connectivity tester.
+	// There might be a delay between IP/DNS configuration being submitted
+	// and having a full effect on the connectivity testing.
+	// For example, the Go resolver reloads resolv.conf at most once per 5 seconds.
+	// See: https://github.com/golang/go/blob/release-branch.go1.18/src/net/dnsclient_unix.go#L362-L366
```

Comment:
How did we avoid issues due to this in the past? I recall having run into this issue a few years ago but I don't remember if we had something to work around the 5 second delay.

Reply:
Pre-refactoring we didn't watch for DNS config being updated by dhcpcd (now we do using fsnotify). This means, that previously nim would react to an updated resolv.conf with a delay through one of the timer handlers (e.g. pendTimer). As a result, there was usually quite a delay between DNS config being available and nim running a connectivity test - enough for Go resolver to update its cache. But still one could experience the same issue if timer fired shortly after DNS config update.
Now we trigger test as soon as we get notified about updated DNS config. This turned out to be a bit too aggressive sometimes, so we need to check what kind of error connection testing has returned and act accordingly.

Reply:
Thanks for the explanation. Makes sense.